### ISSUE TITLE
BraintreeFragment to child FragmentManager

### ISSUE THREAD

**Issue by moia-zsolt** [braintree/braintree_android#252]
### General information

* SDK/Library version: 2.22.0
* Environment: all
* Android Version and Device: n/a
* Braintree dependencies: 
- com.braintreepayments.api:braintree:2.22.0

### Issue description

Our application in general works with representing scenes with `Fragments` which are hosted by a single activity. As the customer navigates in the app, the scene `Fragments` are replaced in the Activity.

One of such `Fragment` is our 'checkout experience'. On this `Fragment`, we take care of using `BraintreeFragment`. The SDK only allows attaching the `BraintreeFragment` to the `FragmentManager` of the host `Activity`. As a result, when the customer leaves the checkout scene, the `BraintreeFragment is still there.`

We'd like to be able to attach `BraintreeFragment` to `Fragment.getChildFragmentManager` so that we don't have to be concerned about what remains there on the `Activity` when the customer leaves the checkout scene.

* Question 1) Would you be able to provide a `BraintreeFragment.newInstance` overload that accepts a `Fragment` instead of `Activity` and attaches to the child `FragmentManager`?
* Question 2) Would you be able to provide means of removing the attached `BraintreeFragment`? Or shall we simply use `BraintreeFragment.TAG`, still somewhat relying on what `newInstance` does internally?

I found #245 somewhat connected because of https://github.com/braintree/braintree_android/issues/245#issuecomment-467096133

**Comment by quinnjn** [comment]
> Question 1) Would you be able to provide a BraintreeFragment.newInstance overload that accepts a Fragment instead of Activity and attaches to the child FragmentManager?

I think this is a reasonable ask, we can add the overloaded method! We'll do some investigation and get back to you, unless you're willing to open a PR!

I think this would be the signature to solve this request.
```
BraintreeFragment#newInstance(Context applicationContext, FragmentManager fragmentManager, String authorization)
```

> Question 2) Would you be able to provide means of removing the attached BraintreeFragment? Or shall we simply use BraintreeFragment.TAG, still somewhat relying on what newInstance does internally?

I think this is also reasonable, but it may cause some confusion for integrators which means we may not want it added at this time.

BraintreeFragment depends heavily on being recreated payment methods that show UI. An example is PayPal browser switch, switching from the Activity to CCT to authorize PayPal, and then coming back to the merchant activity, intercepting the browser switch result. 

I could see the fragment being accidentally removed and not re-attached for these cross app flows. I can play around with it though and see what happens!

Regardless, you can remove BraintreeFragment by the tag as you wish! Just caution that flows away from your app may depend on BraintreeFragment being available when resuming.

**Comment by moia-zsolt** [comment]
> unless you're willing to open a PR!

I definitely want to. (I expect early next week.)

**Comment by zsoltvilagos** [comment]
@quinnjn Bit later but here's the pull request: https://github.com/braintree/braintree_android/pull/253

**Comment by quinnjn** [comment]
Released in 3.1.0. Thanks again for the PR!

**Comment by zsoltvilagos** [comment]
Was my pleasure. Thx for the release.
